crates/bevy_math/src/rotation2d.rs

```diff
@@ -0,0 +1,380 @@
+    pub const FRAC_PI_3: Self = Self {
+        cos: 0.5,
+        sin: 0.866_025_4,
+    };
+
+    /// A counterclockwise rotation of π/4 radians.
+    pub const FRAC_PI_4: Self = Self {
+        cos: std::f32::consts::FRAC_1_SQRT_2,
+        sin: std::f32::consts::FRAC_1_SQRT_2,
+    };
+
+    /// A counterclockwise rotation of π/6 radians.
+    pub const FRAC_PI_6: Self = Self {
+        cos: 0.866_025_4,
+        sin: 0.5,
+    };
+
+    /// A counterclockwise rotation of π/8 radians.
+    pub const FRAC_PI_8: Self = Self {
+        cos: 0.923_879_5,
+        sin: 0.382_683_43,
+    };
+
+    /// Creates a [`Rotation2d`] from a counterclockwise angle in radians.
+    #[inline]
+    pub fn radians(radians: f32) -> Self {
+        #[cfg(feature = "libm")]
+        let (sin, cos) = (
+            libm::sin(radians as f64) as f32,
+            libm::cos(radians as f64) as f32,
+        );
+        #[cfg(not(feature = "libm"))]
+        let (sin, cos) = radians.sin_cos();
+
+        Self::from_sin_cos(sin, cos)
+    }
+
+    /// Creates a [`Rotation2d`] from a counterclockwise angle in degrees.
+    #[inline]
+    pub fn degrees(degrees: f32) -> Self {
+        Self::radians(degrees.to_radians())
+    }
+
+    /// Creates a [`Rotation2d`] from the sine and cosine of an angle in radians.
+    ///
+    /// The rotation is only valid if `sin.asin() == cos.acos()`.
+    #[inline]
+    pub const fn from_sin_cos(sin: f32, cos: f32) -> Self {
+        Self { sin, cos }
+    }
+
+    /// Returns the rotation in radians in the `(-pi, pi]` range.
+    #[inline]
+    pub fn as_radians(self) -> f32 {
+        #[cfg(feature = "libm")]
+        {
+            libm::atan2(self.sin as f64, self.cos as f64) as f32
+        }
+        #[cfg(not(feature = "libm"))]
+        {
+            f32::atan2(self.sin, self.cos)
+        }
+    }
+
+    /// Returns the rotation in degrees in the `(-180, 180]` range.
+    #[inline]
+    pub fn as_degrees(self) -> f32 {
+        self.as_radians().to_degrees()
+    }
+
+    /// Returns the sine and cosine of the rotation angle in radians.
+    #[inline]
+    pub const fn sin_cos(self) -> (f32, f32) {
+        (self.sin, self.cos)
+    }
+
+    /// Returns `true` if the rotation is neither infinite nor NaN.
+    #[inline]
+    pub fn is_finite(self) -> bool {
+        self.sin.is_finite() && self.cos.is_finite()
+    }
+
+    /// Returns `true` if the rotation is NaN.
+    #[inline]
+    pub fn is_nan(self) -> bool {
+        self.sin.is_nan() || self.cos.is_nan()
+    }
+
+    /// Returns `true` if the rotation is near [`Rotation2d::IDENTITY`].
+    #[inline]
+    pub fn is_near_identity(self) -> bool {
+        // TODO: We might be able to use the rotation sine and cosine
+        //       directly instead of first converting to radians.
+        // Same as `Quat::is_near_identity`
+        let threshold_angle = 0.002_847_144_6;
+        self.as_radians().abs() < threshold_angle
```

Comment:
This should work if you just check that:
```suggestion
        cos > 0.0 && sin.abs() < threshold_angle.sin()
```
`threshold_angle.sin()` is of course just another constant.

Conceptually the idea is to cut out a rectangle out of the unit circle, such that only the desired arc remains.

(Please test this suggestion I'm not 100% sure it's entirely correct)